x-pack/legacy/plugins/reporting/server/config/index.ts

```diff
@@ -56,131 +54,6 @@ export interface ReportingConfig extends Config<ReportingConfigType> {
   kbnConfig: Config<KbnServerConfigType>;
 }
 
-type BrowserType = 'chromium';
-
-interface BrowserConfig {
-  inspect: boolean;
-  userDataDir: string;
-  viewport: { width: number; height: number };
-  disableSandbox: boolean;
-  proxy: {
-    enabled: boolean;
-    server?: string;
-    bypass?: string[];
-  };
-}
-
-interface CaptureConfig {
-  browser: {
-    type: BrowserType;
-    autoDownload: boolean;
-    chromium: BrowserConfig;
-  };
-  maxAttempts: number;
-  networkPolicy: NetworkPolicy;
-  loadDelay: number;
-  timeouts: {
-    openUrl: number;
-    waitForElements: number;
-    renderComplete: number;
-  };
-  viewport: any;
-  zoom: any;
-}
-
-interface QueueConfig {
-  indexInterval: string;
-  pollEnabled: boolean;
-  pollInterval: number;
-  pollIntervalErrorMultiplier: number;
-  timeout: number;
-}
-
-interface ScrollConfig {
-  duration: string;
-  size: number;
-}
-
-export interface ReportingConfigType {
-  capture: CaptureConfig;
-  csv: {
-    scroll: ScrollConfig;
-    enablePanelActionDownload: boolean;
-    checkForFormulas: boolean;
-    maxSizeBytes: number;
-    useByteOrderMarkEncoding: boolean;
-  };
-  encryptionKey: string;
-  kibanaServer: any;
-  index: string;
-  queue: QueueConfig;
-  roles: any;
-}
-
-const addConfigDefaults = (
-  server: Legacy.Server,
-  core: CoreSetup,
-  baseConfig: ReportingConfigType
-) => {
-  // encryption key
-  let encryptionKey = baseConfig.encryptionKey;
-  if (encryptionKey === undefined) {
-    server.log(
-      ['reporting', 'config', 'warning'],
-      i18n.translate('xpack.reporting.selfCheckEncryptionKey.warning', {
-        defaultMessage:
-          `Generating a random key for {setting}. To prevent pending reports ` +
-          `from failing on restart, please set {setting} in kibana.yml`,
-        values: {
-          setting: 'xpack.reporting.encryptionKey',
-        },
-      })
-    );
-    encryptionKey = crypto.randomBytes(16).toString('hex');
-  }
-
-  const { kibanaServer: reportingServer } = baseConfig;
-  const serverInfo = core.http.getServerInfo();
-
-  // kibanaServer.hostname, default to server.host, don't allow "0"
-  let kibanaServerHostname = reportingServer.hostname ? reportingServer.hostname : serverInfo.host;
-  if (kibanaServerHostname === '0') {
-    server.log(
-      ['reporting', 'config', 'warning'],
-      i18n.translate('xpack.reporting.selfCheckHostname.warning', {
-        defaultMessage:
-          `Found 'server.host: "0"' in settings. This is incompatible with Reporting. ` +
-          `To enable Reporting to work, '{setting}: 0.0.0.0' is being automatically to the configuration. ` +
-          `You can change to 'server.host: 0.0.0.0' or add '{setting}: 0.0.0.0' in kibana.yml to prevent this message.`,
-        values: {
-          setting: 'xpack.reporting.kibanaServer.hostname',
-        },
-      })
-    );
-    kibanaServerHostname = '0.0.0.0';
```

Comment:
This didn't need to handle the default for `disableSandbox`, because that default was being set in the Joi declaration of the legacy schema.

The `createConfig$` function, in our NP plugin, replaces this function and does handle getting a default for `disableSandbox`